qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -166,8 +166,8 @@ def fit(self, X: np.ndarray, y: np.ndarray):  # pylint: disable=invalid-name
         )
         return super().fit(X, y)
 
-    def _get_interpret(self, num_classes):
-        def parity(x, num_classes=num_classes):
-            return f"{x:b}".count("1") % num_classes
+    def _get_interpret(self, num_classes: int):
+        def parity(x: int, num_classes: int = num_classes) -> int:
+            return x % num_classes
```

Comment:
Can you give an example when this definition of the parity function is better? It is explained in the reno, but I'm not quite sure I follow the explanation.

Reply:
say the number of classes is 4 and the number of output qubits is 2.

then the old parity function would give the following mapping:
```
0 -> "00" -> 0
1 -> "01" -> 1
2 -> "10" -> 1
3 -> "11" -> 2
```
so we only ever map to 0, 1 or 2, which means that later on when we construct the probabilities array (which will have 4 columns) the last column will be all 0, and a 0 gives `nan` in the CE computation.

the new parity function gives this mapping:
```
0 -> 0
1 -> 1
2 -> 2
3 -> 3
```
so that problem described above doesn't exist.

the comment in the reno was to say that even with the new parity function you can still end up with a 0.. unlikely for an entire column, but for individual cells in that column (if a class just has a low probability and we happen to never sample it). so even with this change we still need to fix the case where we take log(0)

Reply:
Thanks for the explanation. I made some additional calculations and realized that the new function yields more uniform results across classes/labels, especially it is noticeable when number of classes increases. Maybe multiple classes should be handled completely differently, e.g. one vs rest or one vs one. But this is another story.